samples/CMakeLists.txt

```diff
@@ -95,83 +73,69 @@ if (CONFIG_SECURE_BOOT)
+
+  elseif(CONFIG_NCS_CHILD_IMAGE_SAMPLE_HCI_RPMSG)
+    set(CHILD_IMAGE_NAME "hci_rpmsg")
+    set(CHILD_IMAGE_PATH "${ZEPHYR_BASE}/samples/bluetooth/hci_rpmsg")
+
+  elseif(CONFIG_NCS_CHILD_IMAGE_SAMPLE_802154_RPMSG)
+    set(CHILD_IMAGE_NAME "802154_rpmsg")
+    set(CHILD_IMAGE_PATH "${ZEPHYR_BASE}/samples/boards/nrf/ieee802154/802154_rpmsg")
+
+  elseif(CONFIG_NCS_CHILD_IMAGE_SAMPLE_RPC_HOST)
+    set(CHILD_IMAGE_NAME "rpc_host")
+    set(CHILD_IMAGE_PATH "${NRF_DIR}/samples/bluetooth/rpc_host")
 
-    add_child_image(
-      NAME ${NETCORE_IMAGE}
-      SOURCE_DIR ${NETCORE_IMAGE_PATH}
-      DOMAIN CPUNET
-      BOARD ${CONFIG_DOMAIN_CPUNET_BOARD})
   else()
-    message(FATAL_ERROR
-      "SoC not supported for including hci_rpmsg as child image")
+    set(CHILD_IMAGE_NAME "hello_world")
+    set(CHILD_IMAGE_PATH "${ZEPHYR_BASE}/samples/hello_world")
   endif()
-elseif (CONFIG_NRF_802154_SER_HOST)
-  if (CONFIG_SOC_NRF5340_CPUAPP)
-    message("Adding '802154_rpmsg' firmware as child image since "
-    "CONFIG_NRF_802154_SER_HOST is set to y")
 
-    add_child_image(
-      NAME 802154_rpmsg
-      SOURCE_DIR ${ZEPHYR_BASE}/samples/boards/nrf/ieee802154/802154_rpmsg
-      DOMAIN CPUNET
-      BOARD ${CONFIG_DOMAIN_CPUNET_BOARD}
-    )
+  if(CONFIG_SOC_NRF5340_CPUAPP)
+    set(CHILD_IMAGE_DOMAIN "CPUNET")
   else()
-    message(FATAL_ERROR
-      "Board ${BOARD} not supported for including '802154_rpmsg' as child image")
+    set(CHILD_IMAGE_DOMAIN "CPUAPP")
   endif()
-endif()
-
-if (CONFIG_NCS_SAMPLE_EMPTY_APP_CORE_CHILD_IMAGE)
-  message("Adding 'empty_app_core' firmware as child image since "
-    "CONFIG_NCS_SAMPLE_EMPTY_APP_CORE_CHILD_IMAGE is set to 'y'")
-
-  add_child_image(
-    NAME empty_app_core
-    SOURCE_DIR ${NRF_DIR}/samples/nrf5340/empty_app_core
-    DOMAIN CPUAPP
-    BOARD ${CONFIG_DOMAIN_CPUAPP_BOARD})
-endif()
 
-if (CONFIG_BT_RPC)
-  if (CONFIG_SOC_NRF5340_CPUAPP)
-    message("Adding 'rpc_host' firmware as child image since "
-      "CONFIG_BT_RPC is set to 'y'")
+  if (CONFIG_NRF53_UPGRADE_NETWORK_CORE)
+    # Inject this configuration from parent image to enable
+    # secure bootloader on the network core. This enables firmware update
+    # of the network core application.
+    add_overlay_config(
+      ${CHILD_IMAGE_NAME}
+      "${NRF_DIR}/subsys/bootloader/image/secure_boot.conf"
+    )
 
-    network_core_firmware_update_enable(rpc_host)
+    # Inject this configuration from parent image mcuboot to enable
+    # the PCD subsystem which is used to communicate the firmware update
+    # to the network core bootloader.
+    add_overlay_config(
+      mcuboot
+      "${NRF_DIR}/subsys/pcd/pcd.conf"
+    )
+  endif()
 
-    add_child_image(
-      NAME rpc_host
-      SOURCE_DIR ${NRF_DIR}/samples/bluetooth/rpc_host
-      DOMAIN CPUNET
-      BOARD ${CONFIG_DOMAIN_CPUNET_BOARD}
+  # Softdevice controller-specific configuration overlay
+  if(CONFIG_BT_RPMSG_NRF53 AND CONFIG_BT_LL_SOFTDEVICE_DEFAULT)
```

Comment:
I am not sure if we still need this overlay file since we have a possibility to add child image overlay files in samples. This overlay was used in some cases before we have this feature also `CONFIG_BT_LL_SOFTDEVICE_DEFAULT` config is really confusing because it doesn't enable the SD controller by default. This config is used still in some thread samples but probably in wrong way since it doesn't enable any controller.
```
# Enable Bluetooth LE Controller that supports multiprotocol
CONFIG_BT_LL_SOFTDEVICE_DEFAULT=y
```

I would suggest removing adding bt_ll_softdevice_hci_rpmsg.conf as overlay here but I would like to have seconds opinion here. What do you think @tejlmand ?

Reply:
yeah the name kinda bugs me too, also I think this overlay wasn't included in all cases as you said. I'd be happy to remove it if you pointed out which samples really depend on it.

Reply:
I went and checked and it seems that the options are either now set by default, or have a child_image/ prj.conf to set them. We'll see if there are implicit requirements when CI runs, but it does seem like this option can be retired. See https://github.com/nrfconnect/sdk-nrf/pull/5052/commits/aebebdf261f8f24fad038a128e1a3423154afb91